File: arrays&hashing/1.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
  const spares = {};
  for (const i in nums) {
    const diff = target - nums[i];
    if (diff in spares) return [i, spares[diff]];
    spares[nums[i]] = i;
  }
};

File: 1.arrays&hashing/1.py
# https://leetcode.com/problems/two-sum/submissions/
# Time: O(n)
# Space: O(n)
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        prevNums = {}
        for i in range(len(nums)):
            diff = target - nums[i]
            if diff in prevNums:
                return [prevNums[diff], i]
            prevNums[nums[i]] = i
